packages/@ember/object/type-tests/try-set.test.ts

```diff
@@ -0,0 +1,16 @@
+import { trySet } from '@ember/object';
+import { expectTypeOf } from 'expect-type';
+
+let obj = { name: 'Zoey' };
+
+let result = trySet(obj, 'contacts.twitter', '@emberjs');
+expectTypeOf(result).toEqualTypeOf<string>();
```

Comment:
Per my note above about this API:

```suggestion
expectTypeOf(result).toEqualTypeOf<string | undefined>();
```